### ISSUE TITLE
Easy way to make the CLI script fail on any errors

### ISSUE THREAD

**Issue by njam** [nuxt-community/nuxt-generate-cluster#16]
### What problem does this feature solve?

When using `nuxt-generate-cluster` in CI it's crucial to detect errors to be able to abort the build process. Otherwise corrupt builds can get deployed.

Currently the CLI script exits with code 0 in many error cases, which makes it difficult to detect that something went wrong.

For example, when a worker process is killed due to hitting NodeJS's memory limit with signal SIGBART, the 'nuxt-generate' script exits with code 0.

<details>

```
$ NODE_OPTIONS=--max-old-space-size=200 nuxt-generate -w1
[...]
✔ generated: /de/events/basel/2019-01-26/5a97ec47-aef9-422c-a75f-db35429a6d84/index.html                                worker 1 18:50:44
✔ generated: /en/events/basel/2019-01-26/5a97ec47-aef9-422c-a75f-db35429a6d84/index.html                                worker 1 18:50:44
✔ generated: /fr/events/basel/2019-01-26/457c9d63-eec8-47b3-9c73-9ffcf5029797/index.html                                worker 1 18:50:44

<--- Last few GCs --->

[26057:0x55c3494ad3b0]     8054 ms: Mark-sweep 189.6 (213.7) -> 182.0 (214.2) MB, 35.8 / 0.0 ms  (average mu = 0.389, current mu = 0.391) allocation failure scavenge might not succeed
[26057:0x55c3494ad3b0]     8113 ms: Mark-sweep 189.8 (214.2) -> 182.2 (214.7) MB, 34.7 / 0.0 ms  (average mu = 0.396, current mu = 0.403) allocation failure scavenge might not succeed


<--- JS stacktrace --->

==== JS stack trace =========================================

    0: ExitFrame [pc: 0x1b9474ddbe1d]
Security context: 0x14ef8771e6e1 <JSObject>
    1: push [0x14ef87705851](this=0x024697cbac39 <JSArray[0]>,0x2534143843c9 <String[1]: ->)
    2: intoTokens [0x1f14ee734521] [/home/njam/Repos/denkmal-website/node_modules/clean-css/lib/tokenizer/tokenize.js:~80] [pc=0x1b94751efc0e](this=0x2290dd79ad11 <JSGlobal Object>,source=0x024697c85b41 <Very long string[1648]>,externalContext=0x024697c86261 <Object ...

FATAL ERROR: Ineffective mark-compacts near heap limit Allocation failed - JavaScript heap out of memory
 1: 0x55c348323f21 node::Abort() [/usr/bin/node]
 2: 0x55c348323f6f  [/usr/bin/node]
 3: 0x55c3484d87f2 v8::Utils::ReportOOMFailure(v8::internal::Isolate*, char const*, bool) [/usr/bin/node]
 4: 0x55c3484d8a4b v8::internal::V8::FatalProcessOutOfMemory(v8::internal::Isolate*, char const*, bool) [/usr/bin/node]
 5: 0x55c34886d283  [/usr/bin/node]
 6: 0x55c34886d3c4  [/usr/bin/node]
 7: 0x55c34887d043 v8::internal::Heap::PerformGarbageCollection(v8::internal::GarbageCollector, v8::GCCallbackFlags) [/usr/bin/node]
 8: 0x55c34887d9c9 v8::internal::Heap::CollectGarbage(v8::internal::AllocationSpace, v8::internal::GarbageCollectionReason, v8::GCCallbackFlags) [/usr/bin/node]
 9: 0x55c34887fadd v8::internal::Heap::AllocateRawWithLigthRetry(int, v8::internal::AllocationSpace, v8::internal::AllocationAlignment) [/usr/bin/node]
10: 0x55c34887fb32 v8::internal::Heap::AllocateRawWithRetryOrFail(int, v8::internal::AllocationSpace, v8::internal::AllocationAlignment) [/usr/bin/node]
11: 0x55c34884dcd5 v8::internal::Factory::NewFillerObject(int, bool, v8::internal::AllocationSpace) [/usr/bin/node]
12: 0x55c348ac028f v8::internal::Runtime_AllocateInNewSpace(int, v8::internal::Object**, v8::internal::Isolate*) [/usr/bin/node]
13: 0x1b9474ddbe1d
ℹ worker 1 exited with status code null by signal SIGABRT                                                                 master 18:50:46
HTML Files generated in 22.5s

$ echo $?
0
```
</details>

### What does the proposed changes look like?

I would suggest to make the script exit with code 1 on:
- Any worker exited with a code other than 0, or got killed
- Any unhandled rendering errors

@pimlie wdyt?

I think the current behaviour (exit 0 even on critical errors) is quite unexpected. But changing it ofc is a breaking change. Not sure what your policy is?

If you think it makes sense, then I would be happy to prepare a PR

<div align="right"><sub><em>This feature request is available on <a href="https://cmty.app/nuxt">Nuxt</a> community (<a href="https://cmty.app/nuxt/nuxt-generate-cluster/issues/c9">#c9</a>)</em></sub></div>

**Comment by pimlie** [comment]
There was a similar issue that got fixed in Nuxt last week: https://github.com/nuxt/nuxt.js/pull/4806

I think it would be easiest to log the [line](https://github.com/nuxt-community/nuxt-generate-cluster/blob/master/lib/cluster/master.js#L105) `worker 1 exited` with consola.fatal if code != 0 or by a signal, then we can use the [same trick](https://github.com/nuxt/nuxt.js/blob/dev/packages/cli/src/setup.js#L26) as nuxt does by exiting when consola.fatal is called.
We could also intercept consola.errors in the worker and then exit the worker by a non-zero exit code, but all consola calls are already sent from the worker to the master first so I dont see the benefit in that (maybe if we'd want to try to restart the worker when it exits with a non-zero code, but I dont think we should do that as its not KISS).

**Comment by pimlie** [comment]
I have released 2.4.0 which include the changes from your pr. I did rename the cli option to `--fail-on-page-error` and removed the `-f` shortcut (prefer to keep the `-f`available for something like force and as discussed I think I will make failing on unhandled page errors the default in a next major release. Thanks

**Comment by njam** [comment]
Nice, thank you!
